### PR TITLE
Implement new Clipboard API v2

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/4379

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                        // recurse on copied children
+                        if (isNotEmpty(item.getChildren())) {
+                            pasteItemsInternal(siteId, operation, newPath, item.getChildren(), pastedItems);
+                        }
+                        break;
+                    default:
+                        logger.warn("Unsupported clipboard operation {0}", operation);
+                }
+
+                pastedItems.add(newPath);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java:90]: This means you only return the root item, correct?

File: src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                        newPath = contentService.moveContent(siteId, item.getPath(), targetPath);
+                        break;
+                    case COPY:
+                        // RDTMP_COPYPASTE
+                        // CopyContent interface is able to send status and new path yet
+                        newPath = contentService.copyContent(siteId, item.getPath(), targetPath);
+
+                        // recurse on copied children
+                        if (isNotEmpty(item.getChildren())) {
+                            pasteItemsInternal(siteId, operation, newPath, item.getChildren(), pastedItems);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java:83]: Nothing is returned, therefore the children's paths are not being returned.

File: src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java (hunk 1 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java:90]: A list is always returned even if only one item was copied/moved

File: src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java (hunk 2 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java:83]: The children's paths are added to the list (last parameter of the method)

File: src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    }
+
+    public String duplicateItem(String siteId, String path) throws ServiceLayerException, UserNotFoundException {
+        // Get the parent url: for folders & components it's just parent, for pages it's the parent of the parent
+        var filename = getName(path);
+        // TODO: Find a better way to do this?
+        var parentUrl = "index.xml".equals(filename)? getFullPathNoEndSeparator(getFullPathNoEndSeparator(path))
+                                                            : getFullPathNoEndSeparator(path);
+        var item = contentService.getContentItem(siteId, parentUrl);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java]: Set depth and make it limited to reduce cost.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Set depth (see notes).

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**